core/drivers/crypto/se050/adaptors/apis/apdu.c

```diff
@@ -0,0 +1,819 @@
+					       SE05X_RSA_NO_priv,
+					       SE05X_RSA_NO_pubMod,
+					       type,
+					       kSE05x_KeyPart_NA,
+					       rsa_format);
+		if (status != SM_OK) {
+			ret = kStatus_SSS_Fail;
+			goto exit;
+		}
+
+		status = Se05x_API_WriteRSAKey(s_ctx,
+					       NULL,
+					       k_object->keyId,
+					       0,
+					       SE05X_RSA_NO_p,
+					       SE05X_RSA_NO_q,
+					       SE05X_RSA_NO_dp,
+					       SE05X_RSA_NO_dq,
+					       SE05X_RSA_NO_qInv,
+					       key_pair->e,
+					       key_pair->e_len,
+					       SE05X_RSA_NO_priv,
+					       SE05X_RSA_NO_pubMod,
+					       type,
+					       kSE05x_KeyPart_NA,
+					       rsa_format);
+		if (status != SM_OK) {
+			ret = kStatus_SSS_Fail;
+			goto exit;
+		}
+
+		status = Se05x_API_WriteRSAKey(s_ctx,
+					       NULL,
+					       k_object->keyId,
+					       0,
+					       SE05X_RSA_NO_p,
+					       SE05X_RSA_NO_q,
+					       SE05X_RSA_NO_dp,
+					       SE05X_RSA_NO_dq,
+					       SE05X_RSA_NO_qInv,
+					       SE05X_RSA_NO_pubExp,
+					       SE05X_RSA_NO_priv,
+					       key_pair->n,
+					       key_pair->n_len,
+					       type,
+					       kSE05x_KeyPart_NA,
+					       rsa_format);
+
+		if (status == SM_OK)
+			ret = kStatus_SSS_Success;
+	}
+exit:
+	return ret;
+}
+
+/*
+ * @param store
+ * @param k_object
+ * @param key_pair
+ * @param key_pub
+ *
+ * @return sss_status_t
+ */
+sss_status_t se050_key_store_set_ecc_key_bin(sss_se05x_key_store_t *store,
+					     sss_se05x_object_t *k_object,
+					     struct ecc_keypair_bin *key_pair,
+					     struct ecc_public_key_bin *key_pub)
+{
+	SE05x_TransientType_t type = kSE05x_TransientType_Persistent;
+	size_t public_keylen = 0;
+	uint8_t buffer[256] = { 0x04 }; /* tag */
+	uint8_t *public_key = buffer + 1;
+	Se05xSession_t *s_ctx = NULL;
+	Se05xPolicy_t policy = {
+		.value = NULL, .value_len = 0,
+	};
+	sss_status_t ret = kStatus_SSS_Fail;
+	smStatus_t status = SM_NOT_OK;
+
+	if (!k_object || !store || !store->session || (!key_pair && !key_pub))
+		return kStatus_SSS_Fail;
+
+	s_ctx = &store->session->s_ctx;
+	k_object->curve_id = key_pair ? key_pair->curve : key_pub->curve;
+	type = k_object->isPersistant ? kSE05x_TransientType_Persistent :
+				  kSE05x_TransientType_Transient;
+
+	if (k_object->objectType != kSSS_KeyPart_Pair)
+		goto label_public;
+
+	public_keylen = key_pair->x_len + key_pair->y_len + 1;
+	if (public_keylen > sizeof(buffer)) {
+		EMSG("small buffer");
+		goto exit;
+	}
+	memcpy(public_key, key_pair->x, key_pair->x_len);
+	memcpy(public_key + key_pair->x_len, key_pair->y, key_pair->y_len);
+
+	status = Se05x_API_WriteECKey(s_ctx,
+				      &policy,
+				      SE05x_MaxAttemps_UNLIMITED,
+				      k_object->keyId,
+				      (SE05x_ECCurve_t)key_pair->curve,
+				      key_pair->d,
+				      key_pair->d_len,
+				      buffer,
+				      public_keylen,
+				      type,
+				      kSE05x_KeyPart_Pair);
+
+	ret = status == SM_OK ? kStatus_SSS_Success : kStatus_SSS_Fail;
```

Comment:
A plain `if`/`else` would be easier to read. Since this appears several times you could even use a helper function: `return sm_to_sss_status(status);`.